src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,182 @@
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_ref' => $node->generateNodeRef(),
+                'url' => $url,
+                'retry_count' => $command->get('ctx_retries'),
+                'stack_trace' => $e->getTraceAsString(),
+            ]);
+        }
+
+        if ($response === null) {
+            return $command;
+        }
+
+        $inspectionResult = $response->getInspectionResult();
+        $indexStatusResult = $inspectionResult->getIndexStatusResult() ?? null;
+        $ampResult = $inspectionResult->getAmpResult() ?? null;
+
+        $isConclusiveForWeb = $indexStatusResult && ($indexStatusResult->getVerdict() === 'PASS' || $indexStatusResult->getVerdict() === 'FAIL');
+        $isConclusiveForAmp = !$node::schema()->hasField('amp_enabled') || ($ampResult && ($ampResult->getVerdict() === 'PASS' || $ampResult->getVerdict() === 'FAIL'));
+
+        if ($isConclusiveForWeb && $isConclusiveForAmp) {
+            $this->putEvent($retryCommand, $pbjx, $node, 'google', $response);
+            return $command;
+        }
+
+        $retryCommand = clone $command;
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+              
+        if ($retries >= $maxRetries) {
+            $this->logger->error('Number of retries for SEO inspection exceeded maximum.', [
+                'node_ref' => $retryCommand->get('node_ref'),
+                'retries' => $retries,
+                'max_retries' => $maxRetries,
+                'web_result_verdict' => $indexStatusResult ? $indexStatusResult->getVerdict() : 'N/A',
+                'amp_result_verdict' => $ampResult ? $ampResult->getVerdict() : 'N/A',
+                'is_conclusive_for_web' => $isConclusiveForWeb,
+                'is_conclusive_for_amp' => $isConclusiveForAmp,
+            ]);
+            $this->putEvent($retryCommand, $pbjx, $node, 'google', $response);
+        } else {
+            $retryCommand->addToSet('search_engines', ['google']);
+        }
+
+        return $retryCommand;
+    }
+
+
+    private function putEvent(Message $command, Pbjx $pbjx, Message $node, string $searchEngine, ?InspectUrlIndexResponse $response = null): void
+    {
+        $nodeRef = $node->generateNodeRef();
+        $event = SeoInspectedV1::create()
+            ->set('node_ref', $node->generateNodeRef())
+            ->set('search_engine', $searchEngine);
+
+        if ($response !== null) {
```

Comment:
@digitalartcrew with the code as written, `$response` will never be null (can skip this check and make the parameter non-nullable)